src/core_modules/capture-core/components/Pages/Search/SearchPage.container.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+import { connect } from 'react-redux';
+import { SearchPage as SearchPageComponent } from './SearchPage.component';
+import { programCollection } from '../../../metaDataMemoryStores';
+import { TrackerProgram } from '../../../metaData/Program';
+import { withErrorMessageHandler, withLoadingIndicator } from '../../../HOC';
+import type { PropsFromRedux } from './SearchPage.types';
+
+const mapStateToProps = (state: ReduxState): PropsFromRedux => {
+    const { currentSelections, activePage } = state;
+
+    const trackedEntityTypesWithCorrelatedPrograms =
```

Comment:
Maybe we should use memoization on this? For the solution I think I prefer React useMemo now that we're using hooks, but reselect would work also.

Reply:
We want  to memoize it in the container or in the component? Can we use `useMemo` in the container?

Reply:
@JoakimSM isn't it enough that we are using `useMemo` on the `trackedEntityTypesWithCorrelatedPrograms` here? 
https://github.com/dhis2/capture-app/pull/800/files#diff-cd18d8025db724faaf2850624427ce57R71

Reply:
I am already preparing a branch that will include this improvement. Let me know if this is fine with you :)